data_steward/cdr_cleaner/cleaning_rules/deid/conflicting_hpo_state_generalization.py

```diff
@@ -0,0 +1,258 @@
+                                     'src_hpos_to_allowed_states.csv')
+
+            # write this to bigquery.
+            try:
+                _ = client.upload_csv_data_to_bq_table(
+                    self.sandbox_dataset_id, MAP_HPO_ALLOWED_STATES, data_path,
+                    UNIT_MAPPING_TABLE_DISPOSITION)
+
+                LOGGER.info(
+                    f"Created {self.sandbox_dataset_id}.{MAP_HPO_ALLOWED_STATES} and "
+                    f"loaded data from {data_path}")
+
+            except Conflict as c:
+                LOGGER.info(
+                    f"{self.sandbox_dataset_id}.{MAP_HPO_ALLOWED_STATES} Data load encountered conflict: "
+                    f"{c.errors}")
+
+            # List Dataset Content
+            dataset_tables = client.list_tables(self.dataset_id)
+            dataset_table_ids = [table.table_id for table in dataset_tables]
+
+            mapped_tables = [
+                table[9:]
+                for table in dataset_table_ids
+                if table.startswith('_mapping_')
+            ]
+
+            # Make sure all mapped_tables exists
+            check_tables = [
+                table for table in mapped_tables if table in dataset_table_ids
+            ]
+
+            # Make sure check_tables contains person_id field
+            person_id_query = LIST_PERSON_ID_TABLES.render(
+                project_id=self.project_id, dataset_id=self.dataset_id)
+            result = client.query(person_id_query).result()
+
+            person_id_tables = []
+            for row in result:
+                table = row.get('table_name')
+                if table in check_tables:
+                    person_id_tables.append(table)
+
+            # Run UNION DISTINCT query to join person_id_tables
+            sql_statements = []
+            for table in person_id_tables:
+                sql_statements.append(
+                    HPO_ID_NOT_RDR_QUERY.render(dataset_id=self.dataset_id,
+                                                table=table))
+            final_hpo_id_not_rdr_query = ' UNION ALL '.join(sql_statements)
+
+            # Create the mapping table in Sandbox if it does not exist in dataset_table_ids
+            if MAP_TABLE_NAME not in dataset_table_ids:
+                schema = [{
+                    "type": "integer",
+                    "name": "person_id",
+                    "mode": "required",
+                    "description": "the person_id of someone with an ehr record"
+                }, {
+                    "type": "string",
+                    "name": "src_hpo_id",
+                    "mode": "required",
+                    "description": "the src_hpo_id of an ehr record"
+                }]
+                table_name = f"{self.project_id}.{self.sandbox_dataset_id}.{MAP_TABLE_NAME}"
+                table = bigquery.Table(table_name, schema=schema)
+                client.create_table(table, exists_ok=True)
+
+            client(
```

Comment:
You can use `client.query(` here, but this will not work.

Reply:
If 'client.query()' won't work, than what would be the best way to run the insert query in sandbox table?
I was trying to follow the approach in 'cancer_concept_suppression.py -> create_suppression_lookup_table()'.

Reply:
Sorry, I was unclear.  You should use `client.query(`.  You've got `client(` and this won't work.